lib/fs.js

```diff
@@ -2510,9 +2510,15 @@ function watchFile(filename, options, listener) {
   filename = pathModule.resolve(filename);
   let stat;
 
-  if (options === null || typeof options !== 'object') {
+  if (typeof options === 'function') {
+    listener = options;
+    options = kEmptyObject;
+  }
+
+  if (!listener) {
```

Comment:
IIUC, when null, 0, false, or NaN is passed to the listener, the error message won't print the value properly.

```suggestion
  if (listener === undefined) {
```